app/models/case_management/icasework/published_request.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module CaseManagement
+  class Icasework
+    # Adapter for the published requests feed from the iCasework Case Management
+    # System.
+    class PublishedRequest
+      def initialize(request)
+        @request = request
+      end
+
+      def reference
+        request.case_id
+      end
+
+      # There's no attribute that just returns the non-prefixed label, so we
+      # have to strip it ourselves.
+      def title
+        case_details[:case_label].gsub('IR - ', '')
+      end
+
+      # TODO: We can't really store this since the access token is only valid
```

Comment:
Need to store the document ID and do an API lookup:

```ruby
# We don't store the URL
def url
  Icasework::Document.new(response_document_id).url
end

private

def response_document_id
  payload[:documents].first { |document| document[:name].starts_with?('Response') }[:id]
end
```

Reply:
```ruby
class PublishedRequest
  def url
    read_attribute(:url) || case_management.classify.generate_url(self)
  end
end

  module CaseManagement
    class Icasework
      def generate_url(published_request)
        published_request.payload[:documents] # and then iterate through and find the right one
      end
    end
  end
```

Reply:
`CaseManagement.for(case_management).generate_url(self)`

```diff
diff --git a/app/models/case_management.rb b/app/models/case_management.rb
index 61669e9..743896b 100644
--- a/app/models/case_management.rb
+++ b/app/models/case_management.rb
@@ -5,19 +5,21 @@
 #
 module CaseManagement
   def self.current
-    @current ||= "CaseManagement::#{config[:adapter]}".constantize.new
+    @current ||= self.for(config[:adapter])
   end
 
   def self.current=(case_management)
     @current = case_management
   end
 
+  def self.for(adapter)
+    "CaseManagement::#{adapter}".constantize.new
+  end
+
   def self.config
     @config ||= Rails.application.config_for(:case_management)
 
-    "CaseManagement::#{@config[:adapter]}".
-      constantize.
-      configure!(@config[:client_params])
+    self.for(config[:adapter]).configure!(@config[:client_params])
 
     @config
   end
```

Reply:
Resolved by 795480d